src/patches/dns.js

```diff
@@ -0,0 +1,39 @@
+// Lambdas running inside a VPC rely on ENIs. On a cold start attaching the ENI
+// can be quite slow and can mean that some services, like DNS, are not yet
+// functional when the Lambda function begins to execute for the first time.
+// Inserting DNS retry logic gives the function a chance to recover before
+// failing completely.
+//
+// See https://docs.aws.amazon.com/lambda/latest/dg/vpc.html#vpc-configuring
+(function () {
+  const dns = require('dns');
+  const lookup = dns.lookup;
+
+  const DELAY = 1000;
+  const TRIES = 5;
+
+  dns.lookup = function dnsLookupWrapper (hostname, options, callback) {
+    let remaining = TRIES;
+
+    function dnsLookupWrapperResponse (error, address, family) {
+      if (error && error.code === dns.NOTFOUND && --remaining > 0) {
```

Comment:
It looks like some of this is implementing retry logic instead of using an existing library. Is there a reason to avoid some of the retry libraries we've used before?

Reply:
My thinking was twofold. First, using a promise-based library makes it a little more difficult/clunky to maintain the Node API interface. Second, using a dependency for a utility in an injected header like this could significantly increase the footprint of this change.

Reply:
@mdlavin any thoughts on this?

Reply:
If it was me, I'd probably use an external library for the retry logic based on callbacks. I think I understand your points and I can appreciate the trade-offs. I'll approve with the code as is it.